packetery/src/Packetery/Module/Checkout.php

```diff
@@ -340,6 +341,54 @@ public function getCartWeightKg() {
 		return $weight;
 	}
 
+	/**
+	 * Calculates fees.
+	 *
+	 * @return void
+	 */
+	public function calculateFees(): void {
+		$feesApi      = WC()->cart->fees_api();
+		$isCod        = false;
+		$codMethod    = $this->options_provider->getCodPaymentMethod();
+		$chosenMethod = WC()->session->get( 'chosen_payment_method' );
+		if ( null !== $codMethod && ! empty( $chosenMethod ) && $chosenMethod === $codMethod ) {
+			$isCod = true;
+		}
+
+		if ( false === $isCod ) {
+			return;
+		}
+
+		$chosenMethod  = $this->getChosenMethod();
+		$carrierOption = get_option( $chosenMethod );
+
+		if ( ! $carrierOption ) {
+			return;
+		}
+
+		$applicableSurcharge = $this->getCODSurcharge( $carrierOption, $this->getCartPrice() );
+		if ( 0 >= $applicableSurcharge ) {
+			return;
+		}
+
+		$fee = [
+			'id'     => 'packetery-cod-surcharge',
+			'name'   => __( 'codSurcharge', 'packetery' ),
+			'amount' => $applicableSurcharge,
+		];
+
+		$feesApi->add_fee( $fee );
+	}
+
+	/**
+	 * Gets cart price.
+	 *
+	 * @return float
+	 */
+	private function getCartPrice(): float {
+		return (float) WC()->cart->get_subtotal();
```

Comment:
Pokud je tam to typování ze stejných důvodů jako tady https://github.com/Zasilkovna/WooCommerce/blob/5efb8284eec0173f9e1403949f4f1d2f2cd5a360/packetery/src/Packetery/Module/EntityFactory/Order.php#L64-L63 , opatřil bych také komentářem.